Code and documentation cleanup

gaurav opened the pull request:
I cleaned up a little of the Python code, and tried to improve the [phyloreference documentation](https://github.com/gaurav/phylo2owl/blob/038f4b3005ba45ed5662e5ac803165caae4f36a2/examples/trees/journal.pone.0094199.md). I think the representation could be prettier, and I think we should play with it before we commit, but I wanted to get this out there so we could try some variations out.

If we end up wanting something really sophisticated, I could modify PhyloDeRef to produce the HTML necessary -- so we're only limited by our imagination here!

gaurav commented:
@hlapp Done! Check it out at https://github.com/gaurav/phylo2owl/blob/79466a9edfce2642feeef3afa389326d0b031470/examples/trees/journal.pone.0094199.md (take 2)

I think the OWL keywords look pretty good now, but I'd like the names to be more immediately recognizable as names-in-OWL, rather than just words. I tried putting ':' in front of them, but I think that just looks annoying: https://github.com/gaurav/phylo2owl/blob/c22fbb1bf47acc3eb5e45551be23c9ecb186cf36/examples/trees/journal.pone.0094199.md (take 3). So if I had to choose, take 2 is probably the best representation for right now.

We could potentially turn those names into URLs, which on browsers would make them visually distinct and would allow someone to immediately look up "excludes_lineage_to" and other terms. Do you think that'd be useful?

hlapp commented:
> We could potentially turn those names into URLs, which on browsers would make them visually distinct and would allow someone to immediately look up "excludes_lineage_to" and other terms. Do you think that'd be useful?

Yes, but that shouldn't be manual should it? Or what way to do this did you have in mind?

hlapp commented:
@gaurav where are we at with this?

gaurav commented:
This pull request covers two things:
1. How to display phyloreferences [in OWL Manchester syntax](https://github.com/gaurav/phylo2owl/blob/79466a9edfce2642feeef3afa389326d0b031470/examples/trees/journal.pone.0094199.md). This is largely deprecated by PHYX now: it's better to describe phyloreferences in terms of specifiers rather than as the OWL property restriction representing them. However, I've opened an issue so that users can view the OWL property restriction in the Curation Tool if they want to (phyloref/curation-tool#64).
2. A minor code cleanup. Based on our experience with the Curation Tool, I've added [pylint](https://www.pylint.org/) to our test suite and fixed all major Pylint errors. This is what this pull request primarily covers now, so I'll change its description to match that.

I forgot about this pull request because we've put phylo2owl on hold until we can replace it with the Python library being developed as part of the [Clade Ontology](https://github.com/phyloref/clade-ontology). But thanks for flagging this pull request up -- once it's been reviewed and merged, phylo2owl can go properly on ice until it's time to work on it again.

Finally, this was from the bad old days when I would develop in my own fork before pushing to phyloref. I've moved all the other branches into the phyloref repository now. I'll keep this branch on my own fork until it's reviewed and merged, and then I'll delete my fork.

hlapp commented on phylo2owl.py:
```diff
@@ -1,144 +1,165 @@
 
-"""phylo2owl.py: Convert phylogenies into OWL ontologies expressed in RDF/XML."""
+"""
+phylo2owl.py: Convert phylogenies into OWL ontologies expressed in RDF/XML.
```
Why not simply _Convert phylogenies into OWL ontologies_? That the OWL is serialized in RDF/XML seems more like current and likely temporary limitation that something important to its design or purpose.

hlapp commented on phylo2owl.py:
```diff
@@ -1,144 +1,165 @@
+
+    # Step 1. Parse command line arguments
+    cmdline_parser = argparse.ArgumentParser(
+        description="Convert phylogenies into OWL ontologies expressed in RDF/XML."
```
See comment above.

hlapp commented on tests/conftest.py:
```diff
@@ -33,8 +32,8 @@ def paths_by_extension(extension):
+    for filename in os.listdir(EXAMPLES_DIR):
         if fnmatch.fnmatch(filename, "*." + extension):
-            paths.append(examples_dir + "/" + filename)
+            paths.append(EXAMPLES_DIR + "/" + filename)
```
This is guarantees failing cross-platform compatibility. Isn't there a constant for for path separator, or a function that joins paths in the manner appropriate for the OS it's running on?

hlapp commented on tests/test_alt_inputs.py:
```diff
@@ -1,17 +1,21 @@
+    Test all .tre files by comparing them to the corresponding .owl file.
+    """
+
+    #pylint: disable=C0103
```
Can we add what this means, beyond an impenetrable code?
